main/config/settings/.defaults.yml

```diff
@@ -170,3 +170,6 @@ auditQldbName: ${self:custom.settings.namespace}-audit
 # Enable this to create a Service Catalog portfolio and populate workspace products
 # Override and disable this setting if you wish to create your portfolio manually
 createServiceCatalogPortfolio: true
+
+# Version number of current release
+customUserAgent: AwsLabs/SO0144/2.1.2
```

Comment:
Does this need to change with every release? I also wonder if this is the best place to put this and make it configurable given that this file is our environment specific settings and we don't want customers to even change this. What do you think?

Reply:
Yep, this value will need to be updated with new releases. This will ensure the AWS SDK requests have the correct version number. 

Since this is the `.default` config file, I don't expect customers to actually edit the values in this file. They'll create their own environment specific files that will inherit from the `.default` config file. I placed the value here as this seems to be the only spot in which we can pass a value to all SDCs. I'm not as familiar with SWB, so please let me know if there are other places in which I can place a value and have it propagate to other SDCs.

Reply:
I actually had something complex in mind- something along the lines of pulling version from change log and adding it to an SSM parameter as part of a deployment script. Later the parameter can be used for various purposes like displaying the version in UI, using it for user agent or even giving warning to users if they are doing a major version release change in the environment. But it seems like we didn't scope the automation as part of the issue you were working on.